### PR TITLE
Ability to reset password

### PR DESCRIPTION
**Pull request by gferraro**
Can reset password which will send a url which can be used to change password within 24 hours.
Link will not work if it has already been used to change the password

### CODE REVIEW

File: api/api/V1/User.ts [hunk 1]
```diff
@@ -257,4 +258,34 @@ export default function (app: Application, baseUrl: string) {
       euaVersion: config.euaVersion,
     });
   });
+
+  app.patch(
+    `${apiUrl}/changePassword`,
+    validateFields([body("token"), body("password")]),
+    fetchUnauthorizedRequiredUserByResetToken(body("token")),
+    validPasswordOf(body("password")),
```
Comment (hardiesoft) [inline api/api/V1/User.ts]: This `validPasswordOf(body("password"))` should go where `body("password")` is in the `validateFields` function.

File: api/api/V1/AuthenticateUser.ts [hunk 2]
```diff
@@ -223,4 +224,58 @@ export default function (app: Application) {
...
+          deprecatedField(validNameOf(body("username"))),
+          validNameOf(body("userName")),
+          validNameOf(body("nameOrEmail")),
+          body("nameOrEmail").isEmail(),
+          body("email").isEmail(),
+        ],
+        "Missing user name in request"
+      ),
+    ]),
+    fetchUnauthorizedOptionalUserByNameOrEmailOrId(
```
Comment (hardiesoft) [inline api/api/V1/AuthenticateUser.ts:249]: Shouldn't this be `fetchUnauthorizedRequiredUserByNameOrEmailOrId`?  Then no need to check for the user later.

File: api/api/V1/AuthenticateUser.ts [hunk 3]
```diff
@@ -223,4 +224,58 @@ export default function (app: Application) {
       });
     })
   );
+
+  app.post(
```
Comment (hardiesoft) [inline api/api/V1/AuthenticateUser.ts:235]: Missing APIdocs

**Review by hardiesoft — CHANGES_REQUESTED** [review]
Looks pretty good, just a couple of API notes to check

File: api/api/V1/AuthenticateUser.ts (hunk 2 above)
Comment (gferraro) [reply api/api/V1/AuthenticateUser.ts:249]: This is a security thing so no matter what you never know if you have gotten an invalid or valid user, when you try to reset a password. i.e. always want to return 200. I think this is normal?

**Pull request closed without merging**